src/components/Signup/SignupForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { AuthError, signUp } from "aws-amplify/auth";
+import Icon from "components/Icon";
+import { Form } from "components/form";
+import { GENERIC_ERROR_MESSAGE } from "constants/common";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { requiredString } from "schemas/string";
+import { object } from "yup";
+import { Donor, StateSetter } from "./types";
+
+type Props = {
+  donor: Donor;
+  setSignupState: StateSetter;
+  classes?: string;
+};
+
+export default function SignupForm(props: Props) {
+  const { handleError } = useErrorContext();
+  const [isPasswordShown, setIsPasswordShown] = useState(false);
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+  } = useForm({
+    resolver: yupResolver(
+      object({
+        password: requiredString
+          .min(8, ({ min }) => `must have at least ${min} characters`)
+          .matches(/[a-z]/, "must have lowercase letters")
```

Comment:
We should abide by what cognito sets (per our cognito config) (used by [authenticator-ui](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/machines/authenticator/defaultServices.ts) ). Trying the same password in `/sign-in` would fail 
<img width="567" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/14329d90-cdea-4a0c-9b75-a421bdba6efa">

Reply:
Makes sense to align these :+1: Weird that they would narrow the number of allowed characters to just the English alphabet.